src/query.ts

```diff
@@ -222,24 +222,39 @@ class Query {
     operatorOrValue?: Operator | AllowedFilterValueType<T>,
     value?: AllowedFilterValueType<T>
   ): Query {
-    if (isFilter(propertyOrFilter)) {
-      this.entityFilters.push(propertyOrFilter);
-      return this;
-    } else {
+    if (arguments.length > 1) {
       process.emitWarning(
         'Providing Filter objects like Composite Filter or Property Filter is recommended when using .filter'
       );
-      let operator = operatorOrValue as Operator;
-      if (arguments.length === 2) {
-        value = operatorOrValue as AllowedFilterValueType<T>;
-        operator = '=';
+    }
+    switch (arguments.length) {
+      case 1: {
+        if (isFilter(propertyOrFilter)) {
+          this.entityFilters.push(propertyOrFilter);
+        } else {
```

Comment:
I think the `else` branch should be removed here (`operatorOrValue` and `value` are not provided)

Reply:
ok